pyproject.toml

```
@@ -7,3 +7,57 @@ disable_error_code = ["union-attr", "annotation-unchecked", "import-untyped"]
 explicit_package_bases = true
 warn_unused_ignores = true
 exclude = ['tests']
+
+[tool.poetry]
+name = "tagstudio"
+version = "9.3.2"
+packages = [
+  {include = "tagstudio/tag_studio.py"},
+  {include = "tagstudio/src/core"},
+  {include = "tagstudio/src/qt"},
+  {include = "tagstudio/resources"},
+  {include = "tagstudio/__init__.py"}
+]
+
+description = "User-Focused Photo & File Management System"
+readme = "README.md"
+authors = ["Travis Abendshien"]
+license = "GPL-3.0-only"
+repository = "https://github.com/TagStudioDev/TagStudio"
+maintainers = [] # Maybe someone can find a way to export this from GitHub?
+
+[tool.poetry.urls]
+Discord = "https://discord.gg/hRNnVKhF2G"
+
+[tool.poetry.dependencies]
+python = ">=3.11 <3.13"
```

Why not supported python 3.13 out of the box?
```suggestion
python = ">=3.11"
```

---

We can't guarantee that the program will work on 3.13 until it's fully out and all of the dependencies have been updated. 

One of the first mistakes I made when opening this project was releasing it on 3.9 and assuming it would work on anything later and labeling it as >=3.9, only to get a bunch of reports from people saying it didn't work on 3.12. Then by updating to 3.12 I unknowingly locked myself out of several modules such as [Open3D](https://www.open3d.org/docs/release/getting_started.html) which are still only supported on up to 3.11.

Basically I'd rather play it safe and wait for the dependencies, both the ones currently being used and the ones I'm still waiting to be updated, before supporting 3.13.

---

No but how I actually wrote it says: Support python version equal to 3.11 and less than 3.13. So that means only 3.11 and 3.12. This is what you asked right?

In fact poetry doesn't even let you install if you say >=3.11 because you get this error:

```
he current project's supported Python range (>=3.11) is not compatible with some of the required packages Python requirement:
  - syrupy requires Python >=3.8.1,<4, so it will not be satisfied for Python >=4

Because tagstudio depends on syrupy (4.6.1) which requires Python >=3.8.1,<4, version solving failed.
```

And I know there is another dependency that actually doesn't support 3.13.

So it was good how it was supporting only 3.11 and 3.12 and your suggestion will not work.

---

The program actually doesn't work with 3.13. I know there are some dependencies that do not support 3.13 yet.